runtime/stringbuf.c

```diff
@@ -281,6 +281,10 @@ rsRetVal rsCStrAppendStrWithLen(cstr_t *pThis, const uchar* psz, size_t iStrLen)
 	memcpy(pThis->pBuf + pThis->iStrLen, psz, iStrLen);
 	pThis->iStrLen += iStrLen;
 
+	/* reset any cached sz string */
+	free(pThis->pszBuf);
```

Comment:
`pBuf` is a buffer of size `iBufSiz`, that contains a string of length `iStrLen`. Initially `iBufSiz==iStrLen`. `pszBuf` is allocated as size `iStrLen+1`. When you extend `pBuf` the content is unchanged (`iBufSiz` increases but `iStrLen` is the same). There is therefore no need to reset `pszBuf`. 
`pszBuf` contains a cached version of the string in `pBuf`, and it will become stale if the string in `pBuf` is modified. It's only when you actually change the _content_ of `pBuf` that you need to reset `pszBuf`.